test/hotspot/jtreg/compiler/lib/template_framework/Template.java

```diff
@@ -0,0 +1,495 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package compiler.lib.template_framework;
+
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.List;
+
+/**
+ * {@link Template}s are used to generate code, based on {@link Token} which are rendered to {@link String}.
+ *
+ * <p>
+ * A {@link Template} can have zero or more arguments, and for each number of arguments there is an implementation
+ * (e.g. {@link ZeroArgs} for zero arguments and {@link TwoArgs} for two arguments). This allows the use of Generics
+ * for the template argument types, i.e. the template arguments can be type checked. Ideally, we would have used
+ * String Templates to inject these arguments into the strings. But since String Templates are not (yet) available,
+ * the {@link Template}s provide <strong>hashtag replacements</strong> in the Strings: the {@link Template} argument
+ * names are captured, and the argument values automatically replace any {@code "#name"} in the Strings. See the
+ * different overloads of {@link make} for examples. Additional hashtag replacements can be defined with {@link let}.
```

Comment:
General Javadocs comment: You should preceed method names with `#` in order to create a proper link.

Without:
![image](https://github.com/user-attachments/assets/b53280a4-8d5a-40a6-84a6-07fc78aa0c2d)

With (i.e. `{@link make}` and `{@link #let})`:
![image](https://github.com/user-attachments/assets/1722614d-9535-410f-b2de-494ecd45a42a)

Reply:
interesting. As we saw offline, `javadoc` does not need the hashtag, but some editors do, and it seems to be common practice to have the hashtag. I'll add them for every applicable link.